sdk/test-utils/perfstress/src/program.ts

```diff
@@ -308,4 +321,63 @@ export class PerfStressProgram {
       }
     }
   }
+
+  /**
+   * This method records the requests-responses and lets the proxy-server know when to playback.
+   * We run runAsync once in record mode to save the requests and responses in memory and then a ton of times in playback.
+   *
+   * ## Workflow of the perf test
+   * - test resources are setup
+   *   - hitting the live service
+   * - then start record
+   *   - making a request to the proxy server to start recording
+   *   - proxy server gives a recording id, we'll use this id to save the actual requests and responses
+   * - run the runAsync once
+   *   - proxy-server saves all the requests and responses in memory
+   * - stop record
+   *   - making a request to the proxy server to stop recording
+   * - start playback
+   *   - making a request to the proxy server to start playback
+   *   - we use the same recording-id that we used in the record mode since that's the only way proxy-server knows what requests are supposed to be played back
+   *   - as a response, we get a new recording-id, which will be used for future playback requests
+   * - run runAsync again
+   *   - based on the duration, iterations, and parallel options provided for the perf test
+   *   - all the requests in the runAsync method are played back since we have already recorded them before
+   * - when the runAsync loops end, stop playback
+   *   - making a request to the proxy server to stop playing back
+   * - delete the live resources that we have created before
+   */
+  private async recordAndStartPlayback(test: PerfStressTest) {
+    // If test-proxy,
+    // => then start record
+    // => run the runAsync
+    // => stop record
+    // => start playback
+    let recorder: TestProxyHttpClientV1 | TestProxyHttpClient;
+    if (PerfStressTest.testProxyHttpClient instanceof TestProxyHttpClient) {
```

Comment:
what would it be other than `instanceof TestProxyHttpClient`? `null`/`undefined`? Couldn't we just check if it exists then?

```suggestion
    if (PerfStressTest.testProxyHttpClient) {
```

Reply:
Should be enough, will do.

Reply:
done